src/job/jobHandler.ts

```diff
@@ -504,12 +564,84 @@ export abstract class JobHandler {
     return await this.executeBuild();
   }
 
+  /* New build process without Makefiles */
+  @throwIfJobInterupted()
+  protected async build(): Promise<boolean> {
+    this.cleanup();
+    const job = this._currJob;
+
+    await this.cloneRepo(this._config.get<string>('repo_dir'));
+    this._logger.save(job._id, 'Cloned Repo');
+
+    await this.commitCheck();
+    this._logger.save(job._id, 'Checked Commit');
+    await this.pullRepo();
+    this._logger.save(job._id, 'Pulled Repo');
+    await this.setEnvironmentVariables();
+    this.logger.save(job._id, 'Prepared Environment variables');
+
+    const buildDependencies = await this.getBuildDependencies();
+    this._logger.save(this._currJob._id, 'Identified Build dependencies');
+
+    const docset = await this._docsetsRepo.getRepo(this._currJob.payload.repoName, this._currJob.payload.directory);
+    let env = this._config.get<string>('env');
+    if (this._currJob.payload.regression) {
+      env = 'regression';
+    }
+    const baseUrl = docset?.url?.[env] || 'https://mongodbcom-cdn.website.staging.corp.mongodb.com';
+
+    const { patchId } = await prepareBuildAndGetDependencies(
+      job.payload.repoName,
+      job.payload.project,
+      baseUrl,
+      buildDependencies,
+      job.payload.directory
+    );
+    this._logger.save(this._currJob._id, 'Downloaded Build dependencies');
+
+    let buildStepOutput: CliCommandResponse;
+
+    const parseFunc = async () => nextGenParse({ job, patchId });
```

Comment:
It appears that the new functions we wrote (e.g. `nextGenParse`, `nextGenHtml`, etc.)  don't throw errors, and we don't check if a step has failed. Because of this, I would assume that the builds will call all of the steps, even if previous ones fail. 

This isn't a huge issue, but I believe we should stop calling subsequent commands if previous ones fail. One thing that this could lead to is incorrect benchmark values as we still save them even if there is an error.